flink-table/flink-sql-parser/src/main/codegen/includes/parserImpls.ftl

```diff
@@ -1773,6 +1773,101 @@ SqlCreate SqlCreateMaterializedTable(Span s, boolean replace, boolean isTemporar
     }
 }
 
+/**
+* Parses alter materialized table.
+*/
+
+SqlAlterMaterializedTable SqlAlterMaterializedTable() :
+{
+    SqlParserPos startPos;
+    SqlIdentifier tableIdentifier;
+    SqlNodeList propertyList = SqlNodeList.EMPTY;
+    SqlNodeList propertyKeyList = SqlNodeList.EMPTY;
+    SqlNodeList partSpec = null;
+    SqlNode freshness = null;
+}
+{
+    <ALTER> <MATERIALIZED> <TABLE> { startPos = getPos();}
+    tableIdentifier = CompoundIdentifier()
+    (
+        <SUSPEND>
+        {
+            return new SqlAlterMaterializedTableSuspend(startPos, tableIdentifier);
+        }
+        |
+        <RESUME>
+        [ <WITH> propertyList = TableProperties() ]
+            {
+                return new SqlAlterMaterializedTableResume(
+                    startPos,
+                    tableIdentifier,
+                    propertyList);
+            }
+        |
+        <REFRESH>
+        [ <PARTITION> {
+                partSpec = new SqlNodeList(getPos());
+                PartitionSpecCommaList(partSpec);
+            }
+        ]
+            {
+                return new SqlAlterMaterializedTableRefresh(
+                    startPos.plus(getPos()),
+                    tableIdentifier,
+                    partSpec);
+            }
+        |
+        <SET>
+        (
+            <FRESHNESS> <EQ> freshness = Expression(ExprContext.ACCEPT_NON_QUERY) {
+                if (!(freshness instanceof SqlIntervalLiteral)) {
+                    throw SqlUtil.newContextException(
+                        getPos(),
+                        ParserResource.RESOURCE.unsupportedFreshnessType());
```

Comment:
If you want to reuse this exception message, I think we should modify ParserResource#unsupportedFreshnessType message to: "MATERIALIZED TABLE only supports define interval type FRESHNESS.",  what do you think?